src/System.Collections.NonGeneric/tests/HashtableTests.cs

```diff
@@ -747,27 +749,27 @@ public static void TestCopyTo_EmptyHashtable()
         }
 
         [Fact]
-        public static void TestCopyTo_Invalid()
+        public static void CopyTo_Invalid()
         {
             Hashtable hash1 = Helpers.CreateIntHashtable(10);
             Helpers.PerformActionOnAllHashtableWrappers(hash1, hash2 =>
             {
-                Assert.Throws<ArgumentNullException>(() => hash2.CopyTo(null, 0)); // Array is null
+                Assert.Throws<ArgumentNullException>("array", () => hash2.CopyTo(null, 0)); // Array is null
 
-                Assert.Throws<ArgumentException>(() => hash2.CopyTo(new object[10, 10], 0)); // Array is multidimensional
+                Assert.Throws<ArgumentException>("array", () => hash2.CopyTo(new object[10, 10], 0)); // Array is multidimensional
                 Assert.Throws<InvalidCastException>(() => hash2.CopyTo(new object[10][], 0)); // Array is invalid
 
-                Assert.Throws<ArgumentOutOfRangeException>(() => hash2.CopyTo(new object[10], -1)); // Index < 0
-                Assert.Throws<ArgumentException>(() => hash2.CopyTo(new object[9], 0)); // Hash.Count + index > array.Length
-                Assert.Throws<ArgumentException>(() => hash2.CopyTo(new object[11], 2)); // Hash.Count + index > array.Length
-                Assert.Throws<ArgumentException>(() => hash2.CopyTo(new object[0], 0)); // Hash.Count + index > array.Length
+                Assert.Throws<ArgumentOutOfRangeException>("arrayIndex", () => hash2.CopyTo(new object[10], -1)); // Index < 0
+                Assert.Throws<ArgumentException>(null,() => hash2.CopyTo(new object[9], 0)); // Hash.Count + index > array.Length
+                Assert.Throws<ArgumentException>(null, () => hash2.CopyTo(new object[11], 2)); // Hash.Count + index > array.Length
+                Assert.Throws<ArgumentException>(null, () => hash2.CopyTo(new object[0], 0)); // Hash.Count + index > array.Length
```

Comment:
I'm curious what the benefit is to adding the parameter name checks when they're null.